tripy/tests/test_function_registry.py

```diff
@@ -270,3 +266,238 @@ def func(**kwargs: Dict[str, Any]):
             return sum(kwargs.values())
 
         assert registry["test"](a=1.0, b=2.0, c=3.0) == 6.0
+
+    def test_sequence_check(self, registry):
+        @registry("test")
+        def func(int_seq: Sequence[int]) -> int:
+            return sum(int_seq)
+
+        assert registry["test"]([1, 2, 3]) == 6
+        # empty should work too
+        assert registry["test"]([]) == 0
+
+    def test_sequence_no_arg_check(self, registry):
+        @registry("test")
+        def func(seq: Sequence) -> int:
+            return len(seq)
+
+        assert registry["test"]([1, 2, 3]) == 3
+        assert registry["test"](["a", "b"]) == 2
+        assert registry["test"]([]) == 0
+
+    def test_union_check(self, registry):
+        @registry("test")
+        def func(n: Union[int, Sequence[int]]) -> int:
+            if isinstance(n, int):
+                return n
+            return sum(n)
+
+        assert registry["test"]([1, 2, 3]) == 6
+        assert registry["test"](6) == 6
+
+    def test_nested_sequence_check(self, registry):
+        @registry("test")
+        def func(n: Sequence[Sequence[int]]) -> int:
+            if n and n[0]:
+                return n[0][0]
+            return -1
+
+        assert registry["test"]([[1, 2, 3], [4, 5, 6]]) == 1
+
+    def test_nested_union_and_sequence_check(self, registry):
+        @registry("test")
+        def func(n: Sequence[Union[int, Sequence[int]]]) -> int:
+            if len(n) == 0:
+                return 0
+            if isinstance(n[0], Sequence):
+                return len(n) * len(n[0])
+            return len(n)
+
+        assert registry["test"]([]) == 0
+        assert registry["test"]([1, 2, 3]) == 3
+        assert registry["test"]([[1, 2], [3, 4], [5, 6]]) == 6
+
+    def test_tensor_literal(self, registry):
+        @registry("test")
+        def func(n: "tripy.TensorLiteral.sig"):
+            return n
+
+        assert registry["test"](1) == 1
+        assert registry["test"](2.0) == 2.0
+        assert registry["test"]([1, 2, 3]) == [1, 2, 3]
+        assert registry["test"]([[0.0, 1.0, 2.0], [3.0, 4.0, 5.0]]) == [[0.0, 1.0, 2.0], [3.0, 4.0, 5.0]]
+
+    def test_error_sequence(self, registry):
+        @registry("test")
+        def func(n: Sequence[int]) -> int:
+            return sum(n)
+
+        with pytest.raises(
```

Comment:
Prefer `helper.raises` over `pytest.raises` since the former will include the exception text in verbose output and can also check for stack information for tensors.